R/tbl_filter.R

```diff
@@ -0,0 +1,146 @@
+#' Filter Hierarchical Tables
+#'
+#' @description `r lifecycle::badge('experimental')`\cr
+#'
+#' This function is used to filter hierarchical table rows. Filters are not applied to summary or overall rows.
+#'
+#' @param x (`tbl_hierarchical`, `tbl_hierarchical_count`)\cr
+#'   A hierarchical gtsummary table of class `'tbl_hierarchical'` or `'tbl_hierarchical_count'`.
+#' @param filter (`expression`)\cr
+#'   An expression that is used to filter rows of the table. See the Details section below.
+#' @param keep_empty (scalar `logical`)\cr
+#'   Logical argument indicating whether to retain summary rows corresponding to table hierarchy sections that have had
+#'   all rows filtered out. Default is `FALSE`.
+#' @inheritParams rlang::args_dots_empty
+#'
+#' @details
+#' The `filter` argument can be used to filter out rows of a table which do not meet the criteria provided as an
+#' expression. Rows can be filtered on the values of any of the possible statistics (`n`, `p`, and `N`) provided they
+#' are included at least once in the table, as well as the values of any `by` variables. Filtering is only applied to
+#' rows that correspond to the innermost variable in the hierarchy - all outer variable (summary) rows are kept
+#' regardless of whether they meet the filtering criteria themselves. In addition to filtering on individual statistic
+#' values, filters can be applied across the row (i.e. across all `by` variable values) by using aggregate functions
+#' such as `sum()` and `mean()`.
+#'
+#' If an overall column was added to the table (via `add_overall())`) this column will not be used in any filters (i.e.
+#' `sum(n)` will not include the overall `n` in a given row). To filter on overall statistics use the `sum()` function
+#' in your filter instead (i.e. `sum(n)` is equal to the overall column `n` across any `by` variables).
+#'
+#' Some examples of possible filters:
+#' - `filter = n > 5`
+#' - `filter = n == 2 & p < 0.05`
+#' - `filter = sum(n) >= 4`
+#' - `filter = mean(n) > 4 | n > 3`
+#' - `filter = any(n > 2 & TRTA == "Xanomeline High Dose")`
+#'
+#' @return A `gtsummary` of the same class as `x`.
+#'
+#' @name tbl_filter
+#' @seealso [tbl_sort()]
+#'
+#' @examplesIf (identical(Sys.getenv("NOT_CRAN"), "true") || identical(Sys.getenv("IN_PKGDOWN"), "true"))
+#' ADAE_subset <- cards::ADAE |>
+#'   dplyr::filter(AEBODSYS %in% c("SKIN AND SUBCUTANEOUS TISSUE DISORDERS",
+#'                                 "EAR AND LABYRINTH DISORDERS")) |>
+#'   dplyr::filter(.by = AEBODSYS, dplyr::row_number() < 20)
+#'
+#' tbl <-
+#'   tbl_hierarchical(
+#'     data = ADAE_subset,
+#'     variables = c(AEBODSYS, AEDECOD),
+#'     by = TRTA,
+#'     denominator = cards::ADSL |> mutate(TRTA = ARM),
+#'     id = USUBJID,
+#'     overall_row = TRUE
+#'   )
+#'
+#' # Example 1 - Row Sums > 3 -------------------
+#' tbl_filter(tbl, sum(n) > 3)
+#'
+#' # Example 2 - Row Sums <= 3 ------------------
```

Comment:
Can we illustrate another filtering method here?

Reply:
Done!